server/handler/v2/handler.go

```diff
@@ -90,8 +91,12 @@ func (b *blobHandler) getBlob(w http.ResponseWriter, r *http.Request) {
 		b.handleError(w, errors.New("key query parameter is required"), http.StatusBadRequest)
 		return
 	}
+	key, err := url.QueryUnescape(keyParam)
```

Comment:
What needs unescaping in our key? The `/` probably, right? 
Looks basically ok to me. Do you mind adding a test. If you don't want to add a new test on the handler level, I think you can just add a test case to `server_test` where we already have a retrieval test. Just add another case where the `/` is escaped.